mne/io/base.py

```diff
@@ -658,7 +658,7 @@ def filenames(self):
         return tuple(self._filenames)
 
     @annotations.setter
-    def annotations(self, annotations):
+    def annotations(self, annotations, emit_warning=True):
```

Comment:
This is for special cases like reading split files and dealing with floating point inaccuracies. Other option would be to set the private attribute, but then the checks done here should be done elsewhere. IMO this is cleaner.